dotnet/test/Azure.Iot.Operations.Services.IntegrationTests/AdrServiceClientIntegrationTests.cs

```diff
@@ -0,0 +1,758 @@
+    }
+
+    [Fact]
+    public async Task ObservationsPersistAfterReconnection()
+    {
+        // Arrange
+        await using MqttSessionClient mqttClient = await ClientFactory.CreateAndConnectClientAsyncFromEnvAsync();
+        ApplicationContext applicationContext = new();
+        await using AdrServiceClient client = new(applicationContext, mqttClient, ConnectorClientId);
+
+        // Set up tracking for received events
+        var receivedEvents = new List<Asset>();
+        var firstEventReceived = new TaskCompletionSource<bool>();
+        var reconnectionEventReceived = new TaskCompletionSource<bool>();
+        var eventCounter = 0;
+
+        // Set up event handler to track events
+        client.OnReceiveAssetUpdateEventTelemetry += (_, asset) =>
+        {
+            if (asset != null)
+            {
+                _output.WriteLine($"Received asset event: {asset.Name}, count: {++eventCounter}");
+                receivedEvents.Add(asset);
+
+                // Signal based on which event we're processing
+                if (eventCounter == 1)
+                {
+                    firstEventReceived.TrySetResult(true);
+                }
+                else if (eventCounter > 1)
+                {
+                    reconnectionEventReceived.TrySetResult(true);
+                }
+            }
+            return Task.CompletedTask;
+        };
+
+        // Start observing asset updates
+        var observeResponse = await client.ObserveAssetUpdatesAsync(
+            TestDevice_1_Name, TestEndpointName, TestAssetName);
+        Assert.Equal(NotificationResponse.Accepted, observeResponse);
+
+        // Act - Phase 1: Send an update and verify it's received
+        var updateRequest1 = CreateUpdateAssetStatusRequest(DateTime.UtcNow);
+        updateRequest1.AssetStatus.Events = new List<AssetDatasetEventStreamStatus>
+        {
+            new AssetDatasetEventStreamStatus
+            {
+                Name = "pre-disconnect-event",
+                MessageSchemaReference = new MessageSchemaReference
+                {
+                    SchemaName = "test-schema",
+                    SchemaRegistryNamespace = "test-namespace",
+                    SchemaVersion = "1.0"
+                }
+            }
+        };
+
+        await client.UpdateAssetStatusAsync(
+            TestDevice_1_Name, TestEndpointName, updateRequest1);
+
+        // Wait for the first event
+        var firstEventReceived1 = await Task.WhenAny(
+            firstEventReceived.Task,
+            Task.Delay(TimeSpan.FromSeconds(5)));
+
+        Assert.True(firstEventReceived1.IsCompleted && !firstEventReceived1.IsFaulted,
+            "Did not receive first asset event update within timeout");
+
+        // Act - Phase 2: Simulate reconnection by disposing and recreating the client
+        _output.WriteLine("Simulating disconnection by disposing client...");
+        await client.DisposeAsync();
```

Comment:
Could you just close and re-open the same client rather than dispose and instantiate a new client? That would be more similar to simulating a reconnection.

Reply:
It would also mean you wouldn't create a new ADR service client